### PR TITLE
Fix `Vector#slice` for arguments near `Int.max/minValue`

### LINKED ISSUE #12739: Vector(1).slice(Int.MaxValue, Int.MinValue) == Vector(1)

### PR DESCRIPTION
**Pull request by som-snytt**
Fixes scala/bug#12739

@scala/collections 

At first I'm like, even I can understand this one, it's not like those Rex Kerr issues. Unless I don't understand it.

### CODE REVIEW

File: test/junit/scala/collection/immutable/VectorTest.scala [hunk 1]
```diff
@@ -163,7 +163,7 @@ class VectorTest {
   }
 
   @Test
-  def testWierdAlignments1(): Unit = {
+  def testWeirdAlignments1(): Unit = {
```
Comment (som-snytt) [inline test/junit/scala/collection/immutable/VectorTest.scala:166]: shoutout to Weird Al!

**Review by SethTisue — APPROVED** [review]
I'm tentatively hitting “Approve”, but I would appreciate at least one more pair of eyes from @scala/collections crew

**Comment by SethTisue** [comment]
perhaps @AminMal would like to take a look

**Comment by AminMal** [comment]
> perhaps @AminMal would like to take a look

Thanks, I'd love to. I took a look and it seems like it actually should've been this way in the first place, and couldn't find a scenario where this change could go wrong. Besides from the actual bug (which would be fixed in this PR), it makes sense to check if the new length is not negative, prior to comparing lengths and creating a slice out of the current vector.

**Comment by som-snytt** [comment]
> Besides from the actual bug (which would be fixed in this PR), it makes sense to check if the new length is not negative, prior to comparing lengths and creating a slice out of the current vector.

I understand the "besides" clause, but not the part that "makes sense".

Rebased.

**Comment by AminMal** [comment]
> > Besides from the actual bug (which would be fixed in this PR), it makes sense to check if the new length is not negative, prior to comparing lengths and creating a slice out of the current vector.
> 
> I understand the "besides" clause, but not the part that "makes sense".
> 
> Rebased.

I mean it makes sense to check that `hi <= lo` before `hi - lo == length`. If we get `Vector(2).slice(10, 3)`, we shouldn't bother checking whether `-7` is equal to `1` or not, we should return an empty vector immediately.

**Comment by som-snytt** [comment]
You're saying the current fix makes sense, not that "in addition, it would make sense to do an additional check"...

English verb tenses and moods are so confusing.

**Comment by AminMal** [comment]
> You're saying the current fix makes sense, not that "in addition, it would make sense to do an additional check"...
> 
> English verb tenses and moods are so confusing.

I get it, I'm not a "native English speaker", so that's why, sorry if it made you confused. All I was saying was that the change in the PR makes sense :)

**Comment by SethTisue** [comment]
thank you both!

**Comment by som-snytt** [comment]
@AminMal thanks again for review. I was just reading wrong what you wrote right.

Possibly I was in a panic that it would turn into a marathon Ichoran-like thread about arithmetic.

**Pull request merged**